sound/soc/sof/topology.c

```diff
@@ -725,6 +746,13 @@ static const struct sof_topology_token core_tokens[] = {
 		offsetof(struct sof_ipc_comp, core), 0},
 };
 
+/* widget */
+static const struct sof_topology_token uuid_tokens[] = {
+	{SOF_TKN_COMP_UUID,
+		SND_SOC_TPLG_TUPLE_TYPE_UUID, get_token_uuid,
+		offsetof(struct sof_ipc_comp, uid), 0},
```

Comment:
Another "uid" here, "user identifier"?

Reply:
I reused the naming from the FW to keep them consistent, because the struct sof_ipc_comp will be used both side and eventually the "uid" will be used inside firmware for comparison.
```
/**
 * Audio component base driver "class"
 * - used by all other component types.
 */
struct comp_driver {
	uint32_t type;		/**< SOF_COMP_ for driver */
	uint32_t uid;		/**< Address of uuid_entry */
	struct tr_ctx *tctx;	/**< Pointer to trace context */
	struct comp_ops ops;	/**< component operations */
};
```